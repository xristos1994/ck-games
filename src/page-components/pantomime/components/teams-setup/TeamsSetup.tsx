import React, { FC, ReactElement } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { classnames, compose } from '@utils/component-utils';
import { withTranslation, ITranslate } from '@models/i18n/hoc';
import { Button } from '@components';
import { teams, isTeamsSetupValid } from '@models/pantomime/props';
import { Team } from './components';
import { addTeam, teamsSetupSubmit } from '@models/pantomime/actions';
import { IState } from '@models/interfaces';

import styles from './styles.module.css';

interface IProps {
  teams: {
    id: number;
    score: number;
    name: string;
  }[];
  teamsSetupSubmit: () => void;
  addTeam: () => void;
  isTeamsSetupValid: boolean;
  t: ITranslate;
}

const _TeamsSetup: FC<IProps> = ({ teams, teamsSetupSubmit, addTeam, isTeamsSetupValid, t }): ReactElement => {
  return (
    <div className={classnames(styles.teamsSetupContainer)}>
      <div className={classnames(styles.teamsSetupTitle)}>{t('Team Setup')}</div>
      <div className={styles.teamsContainer}>
        {teams.map((team) => (
          <Team key={team.id} team={team} />
        ))}
      </div>
      <Button onClick={() => addTeam()} className={classnames(styles.addTeamButton)}>
        {t('Add Team')}
      </Button>

      <Button
        other={{ disabled: !isTeamsSetupValid }}
        onClick={() => teamsSetupSubmit()}
        className={classnames(styles.teamsSetupSubmitButton)}
      >
        {t('CONTINUE')}
      </Button>
    </div>
  );
};

const TeamsSetup = compose(
  connect(
    createStructuredSelector<
      IState,
      {
        teams: IProps['teams'];
        isTeamsSetupValid: IProps['isTeamsSetupValid'];
      },
      {
        teams: IProps['teams'];
        isTeamsSetupValid: IProps['isTeamsSetupValid'];
      }
    >({
      teams,
      isTeamsSetupValid
    }),
    { teamsSetupSubmit, addTeam }
  ),
  withTranslation
)(_TeamsSetup);

export { TeamsSetup };
