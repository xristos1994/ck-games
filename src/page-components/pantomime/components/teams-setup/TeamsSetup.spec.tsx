import React, { MouseEventHandler, ReactChild } from 'react';
import renderer from 'react-test-renderer';
import { _TeamsSetup as TeamsSetup } from './TeamsSetup';

const Button = (
  { children, onClick, className='' } :
  { children: ReactChild | ReactChild[]; onClick: MouseEventHandler<HTMLButtonElement>; className: string }
) => {
  return (
    <button onClick={onClick} className={className || undefined}>
      {children}
    </button>
  );
};

jest.mock('@components', () => ({
  __esModule: true,
  Button: Button
}));

const Team = ({ team, canBeRemoved }: { team: { id: number; name: string }, canBeRemoved: boolean }) => {
  const { id, name } = team;

  return (
    <div>
      {`${id}_${name}_${canBeRemoved}`}
    </div>
  );
};

jest.mock('./components/', () => ({
  __esModule: true,
  Team: Team
}));

const t = (label: string, param?: string[]) => label + (param ? param[0] : '');
const teamsSetupSubmit = () => void 0;
const addTeam = () => void 0;

describe('page-components/Pantomime/TeamsSetup', () => {
  it('renders correctly', () => {
    let tree = renderer.create(
      <TeamsSetup
        isTeamsSetupValid={true}
        teamsSetupSubmit={teamsSetupSubmit}
        addTeam={addTeam}
        teams={[
          { id: 1, name: 'Name 1' },
          { id: 2, name: 'Name 2' },
          { id: 3, name: 'Name 3' }
        ]}
        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        // @ts-ignore
        t={t}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();

    tree = renderer.create(
      <TeamsSetup
        isTeamsSetupValid={false}
        teamsSetupSubmit={teamsSetupSubmit}
        addTeam={addTeam}
        teams={[
          { id: 1, name: 'Name 1' },
          { id: 2, name: '' },
          { id: 3, name: 'Name 3' }
        ]}
        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        // @ts-ignore
        t={t}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
