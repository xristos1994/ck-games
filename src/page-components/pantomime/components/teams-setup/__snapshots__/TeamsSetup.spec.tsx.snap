// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page-components/Pantomime/TeamsSetup renders correctly 1`] = `
<div
  className="teamsSetupContainer"
>
  <div
    className="teamsSetupTitle"
  >
    Team Setup
  </div>
  <div
    className="teamsContainer"
  >
    <div>
      1_Name 1_true
    </div>
    <div>
      2_Name 2_true
    </div>
    <div>
      3_Name 3_true
    </div>
  </div>
  <button
    className="addTeamButton"
    onClick={[Function]}
  >
    Add Team
  </button>
  <button
    className="teamsSetupSubmitButton"
    onClick={[Function]}
  >
    CONTINUE
  </button>
</div>
`;

exports[`page-components/Pantomime/TeamsSetup renders correctly 2`] = `
<div
  className="teamsSetupContainer"
>
  <div
    className="teamsSetupTitle"
  >
    Team Setup
  </div>
  <div
    className="teamsContainer"
  >
    <div>
      1_Name 1_true
    </div>
    <div>
      2__true
    </div>
    <div>
      3_Name 3_true
    </div>
  </div>
  <button
    className="addTeamButton"
    onClick={[Function]}
  >
    Add Team
  </button>
  <button
    className="teamsSetupSubmitButton"
    onClick={[Function]}
  >
    CONTINUE
  </button>
</div>
`;
